### PR TITLE
Mobile login

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: mobile/modules/users/less/_login.less [hunk 1]
```diff
@@ -15,6 +16,74 @@
   height: 100%;
   color: @color-white;
```
Comment (make-it-happen-team) [inline mobile/modules/users/less/_login.less]: could you use component variable (mapped from color)?

File: mobile/modules/users/less/_login.less [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
+@import "../../../less/_app.mixins.less";
 
 .main-content {
-  width: @mobile-width;
-  height: @mobile-height;
+  width: 100%;
+  height: 100vh;
 }
 .user-login {
   padding: @mobile-padding;
```
Comment (make-it-happen-team) [inline mobile/modules/users/less/_login.less:10]: could you use special to login variable?

File: mobile/modules/users/less/_login.less [hunk 3]
```diff
@@ -15,6 +16,74 @@
...
+            height: auto;
+            line-height: 0;
+          }
+        }
+      }
+    }
+  }
+
+  &.user-login--reset-password {
+    color: @color-white;
```
Comment (make-it-happen-team) [inline mobile/modules/users/less/_login.less]: could you use component variable (mapped from color)?

File: mobile/modules/users/less/_login.less [hunk 4]
```diff
@@ -15,6 +16,74 @@
...
+
+    .login-view__form-control.form-control {
+      border-bottom: 1px solid @color-bali-hai-blue-s24;
```
Comment (make-it-happen-team) [inline mobile/modules/users/less/_login.less]: could you use component variable (mapped from color)?

Comment (make-it-happen-team) [reply mobile/modules/users/less/_login.less]: all variables for this module

File: mobile/modules/users/less/_login.less [hunk 5]
```diff
@@ -125,19 +196,43 @@
...
+      }
+
+      &:-moz-placeholder {
+        font-size: 1.4rem;
+        letter-spacing: 0;
+      }
+
+      &:-ms-input-placeholder {
+        font-size: 1.4rem;
+        letter-spacing: 0;
```
Comment (make-it-happen-team) [inline mobile/modules/users/less/_login.less:234]: do we really need selectors for special browsers?

File: mobile/modules/users/views/password/reset-password.client.view.html [hunk 6]
```diff
@@ -1,4 +1,4 @@
-<form class="signin form-horizontal"
+<form class="user-login user-login--reset-password form-horizontal"
```
Comment (make-it-happen-team) [inline mobile/modules/users/views/password/reset-password.client.view.html]: probably we are not  going to use bootstrap on mobile. could we do that without special bootstrap classes?

File: public/modules/users/services/password-validator.client.service.es6 [hunk 7]
```diff
@@ -1,27 +1,28 @@
...
+    this.owaspPasswordStrengthTest.tests.optional = this.owaspPasswordStrengthTest.tests.optional.filter((val) => !(/one special character/).test(val));
+  }
+  getResult (password) {
+    let result    = owaspPasswordStrengthTest.test(password);
 
-		owaspPasswordStrengthTest.tests.optional = owaspPasswordStrengthTest.tests.optional.filter(function(val){
-			return !(/one special character/).test(val);
-		});
+    result.errors = result.errors.map((val) => val.replace(/The password (must|may)/g, '-'));
+    console.log(result);
```
Comment (make-it-happen-team) [inline public/modules/users/services/password-validator.client.service.es6]: please remove redundant log

**Review by make-it-happen-team — CHANGES_REQUESTED** [review]

File: mobile/modules/users/directives/password-validator.client.directive.es6 [hunk 8]
```diff
@@ -2,7 +2,6 @@ angular.module('users')
 .directive('passwordValidator', ['PasswordValidator', "$popover", function(PasswordValidator,$popover ) {
 	return {
```
Comment (make-it-happen-team) [inline mobile/modules/users/directives/password-validator.client.directive.es6:3]: Could we refactor it to es6?

**Review by make-it-happen-team — CHANGES_REQUESTED** [review]

File: mobile/modules/users/less/_login.less (hunk 2 above)
Comment (ghost) [reply mobile/modules/users/less/_login.less:10]: They are removed.

File: mobile/modules/users/less/_login.less (hunk 1 above)
Comment (ghost) [reply mobile/modules/users/less/_login.less]: Fixed

File: mobile/modules/users/less/_login.less (hunk 3 above)
Comment (ghost) [reply mobile/modules/users/less/_login.less]: Fixed

File: mobile/modules/users/less/_login.less (hunk 4 above)
Comment (ghost) [reply mobile/modules/users/less/_login.less]: Fixed

File: mobile/modules/users/less/_login.less (hunk 5 above)
Comment (ghost) [reply mobile/modules/users/less/_login.less:234]: That's the way to style placeholder. We need that

File: mobile/modules/users/views/password/reset-password.client.view.html (hunk 6 above)
Comment (ghost) [reply mobile/modules/users/views/password/reset-password.client.view.html]: Fixed

File: public/modules/users/services/password-validator.client.service.es6 (hunk 7 above)
Comment (ghost) [reply public/modules/users/services/password-validator.client.service.es6]: Fixed

File: mobile/modules/users/directives/password-validator.client.directive.es6 (hunk 8 above)
Comment (ghost) [reply mobile/modules/users/directives/password-validator.client.directive.es6:3]: Yes, but not today.

File: mobile/modules/users/less/_login.less (hunk 5 above)
Comment (make-it-happen-team) [reply mobile/modules/users/less/_login.less:234]: in this case we could take out common styles

&::-webkit-input-placeholder,
&::-moz-placeholder,
etc. {
     font-size: 1.4rem;
     letter-spacing: 0;
}

File: mobile/modules/users/views/password/reset-password.client.view.html (hunk 6 above)
Comment (make-it-happen-team) [reply mobile/modules/users/views/password/reset-password.client.view.html]: form-horizontal?

**Review by make-it-happen-team — CHANGES_REQUESTED** [review]

File: mobile/modules/users/less/_login.less (hunk 5 above)
Comment (ghost) [reply mobile/modules/users/less/_login.less:234]: Unfortunately no. I try your option before. Doesn't work in this way

File: mobile/modules/users/views/password/reset-password.client.view.html (hunk 6 above)
Comment (ghost) [reply mobile/modules/users/views/password/reset-password.client.view.html]: Fixed

**Pull request merged**